src/mac_features/ack_generator/nrf_802154_ack_data.h

```diff
@@ -99,6 +99,20 @@ bool nrf_802154_ack_data_for_addr_clear(const uint8_t * p_addr, bool extended, u
  */
 void nrf_802154_ack_data_reset(bool extended, uint8_t data_type);
 
+/**
+ * @brief Select the source matching algorithm.
+ *
+ * @note This method should only be called in the initialization phase.
+ *
+ * When calling nrf_802154_ack_data_pending_bit_should_be_set method, several different
+ * algorithms can be chosen to determine whether the pending bit should be set.
+ *
+ * @see nrf_802154_src_match_t
```

Comment:
I changed this argument to uint8_t for similarity with other wrappers in this module, as other types defined by nrf_802154_types module are not used in this module as well - uint8_t is used instead. Because of this, the nrf_802154_types module is included by c file and not h file, and those types cannot be used without changing the includes. Should the includes be refactored?

Reply:
Which types defined by `nrf_802154_types` do you refer to?

Reply:
One example could be `nrf_802154_ack_data_t`, which is not directly used anywhere in the module, but the macros `NRF_802154_ACK_DATA_PENDING_BIT` and `NRF_802154_ACK_DATA_IE` which should be its values are used directly in function calls. Arguments of these function on the other hand are `uint8_t`, instead of `nrf_802154_ack_data_t`. One example could be `addr_index_find` function and its `data_type` argument.

Reply:
Ok, I understand now.
I think we should use `nrf_802154_ack_data_t` in this header file and include `nrf_802154_types.h` here and in the .c file as well. It would be clearer approach. Or am I missing some drawbacks of such solution?